# 📚The Briefing Room

## Real example: Find, Clean and Merge Data.
As a Data Analyst, “find, clean and merge” Data is the essential step for your daily role. 
On one occasion, When I was working as Data Analyst and Project Consultant for a state deputy cabinet, the parliament was in a heated moment because of a crisis between several Brazilian states, concerning about the tax rates on jet fuel. 
In one hand was one parliamentary group trying to collect more taxes, and in the other side was a party that advocated the opposite: they were trying to lower the rate, in the attempt to promote more international passengers flights to our state (specially from Europe, because the flights from the US were already consolidated).
As part of my job as a consultant, I needed to provide insights that could help the debate and, at the end, support my team.  
My goal was to find Data and show the situation of the Brazilian international aviation. For that, my idea was to create a dynamic dashboard in Tableau, showing with some lines, the density of flights from Europe to the Northeast region of Brazil. With this data precious asset, I could compare the flights to our state with the others from the same region of Brazil.

## 🔎 Where to find trustful Data and what to do with it?
Whenever possible, try to find Data directly from a government source or from a reference company, only after that try on open private repositories. This is the key part of the job: find good and trustful Data.
After some research, I found a public repository from the National Civil Aviation Agency of Brazil, containing a 220MB .csv file, with all the database of registered flights in Brazil since 2000.
The updated Data (until march 2021) can be found [here]( https://www.anac.gov.br/acesso-a-informacao/dados-abertos/areas-de-atuacao/voos-e-operacoes-aereas/dados-estatisticos-do-transporte-aereo/48-dados-estatisticos-do-transporte-aereo-formato-csv).
The only problem with this dataset is that is missing the coordinates from each airport, what is a trouble for geographic localization in Tableau.
I could not find an official repository with geographic location of the aforementioned airports, so I used this one [here](https://datahub.io/core/airport-codes).
With these two datasets I could merge then, allowing me to have all the flights information added to the coordinates of each airport, from departure to arrival.

## 🧹 🛠 Get your hands dirty
Once you have your Data, it’s time to get your hands dirty. 
The csv file has more than 900K rows and 35 columns, what makes impossible to work with Excel. 
So, to handle this Dataset I used Python, with the two libraries listed below. 
In the next lines the work can be seen, generating, at the end, a clean and smaller csv file, with just the necessary data to be used in Tableau.

-	[X] Pandas
-	[X] Chardet


## ✒️ Developer

Marcelo Pasquini

 [LinkedIn Profile](https://www.linkedin.com/in/mpbrazil/)


In [1]:
import pandas as pd
import chardet

In [2]:
with open("032021_ANAC_Dados_Estatisticos.csv", 'rb') as rawdata:
    result = chardet.detect(rawdata.read(100000))
result

{'encoding': 'ISO-8859-1', 'confidence': 0.73, 'language': ''}

In [3]:
df_original = pd.read_csv("032021_ANAC_Dados_Estatisticos.csv", encoding='ISO-8859-1', sep=";")

In [4]:
df = df_original.copy()
df.sample(5)

EMPRESA (SIGLA)                                     EMPRESA (NOME)  \
140586             VSP                             VIAÇÃO AÉREA SÃO PAULO   
344945             TTL                           TOTAL LINHAS AÉREAS S.A.   
447380             NHG                BRAVA LINHAS AÉREAS LTDA. (EX- NHT)   
218104             VSP                             VIAÇÃO AÉREA SÃO PAULO   
636894             GLO  GOL LINHAS AÉREAS S.A. (EX- VRG LINHAS AÉREAS ...   

       EMPRESA (NACIONALIDADE)   ANO  MÊS AEROPORTO DE ORIGEM (SIGLA)  \
140586              BRASILEIRA  2002    9                        SBFL   
344945              BRASILEIRA  2007    1                        SBRJ   
447380              BRASILEIRA  2010   10                        SBSM   
218104              BRASILEIRA  2004    3                        SBRF   
636894              BRASILEIRA  2014    5                        SBGR   

       AEROPORTO DE ORIGEM (NOME) AEROPORTO DE ORIGEM (UF)  \
140586              FLORIANÓPOLIS                       SC   
344945             RIO DE JANEIRO                       RJ   
447380                SANTA MARIA                       RS   
218104                     RECIFE                       PE   
636894                  GUARULHOS                       SP   

       AEROPORTO DE ORIGEM (REGIÃO) AEROPORTO DE ORIGEM (PAÍS)  ...  \
140586                          SUL                     BRASIL  ...   
344945                      SUDESTE                     BRASIL  ...   
447380                          SUL                     BRASIL  ...   
218104                     NORDESTE                     BRASIL  ...   
636894                      SUDESTE                     BRASIL  ...   

       COMBUSTÍVEL (LITROS) DISTÂNCIA VOADA (KM) DECOLAGENS CARGA PAGA KM  \
140586                  NaN                  NaN        NaN           NaN   
344945               1503.0                542.0        1.0           0.0   
447380               6560.0               6460.0       20.0           0.0   
218104                  NaN                  NaN        NaN           NaN   
636894             680738.0             168960.0       88.0    58168300.0   

       CARGA GRATIS KM CORREIO KM ASSENTOS    PAYLOAD HORAS VOADAS  \
140586             NaN        NaN      NaN        NaN          NaN   
344945             0.0        0.0     48.0     3800.0         2,37   
447380             0.0        0.0    380.0    23002.0        25,45   
218104             NaN        NaN      NaN        NaN          NaN   
636894        545280.0        0.0  15609.0  1401010.0       238,42   

        BAGAGEM (KG)  
140586          34.0  
344945         309.0  
447380          96.0  
218104         209.0  
636894      146896.0  

[5 rows x 38 columns]

In [5]:
df.rename(columns={'ANO':'YEAR', 'MÊS':'MONTH'}, inplace=True)

In [6]:
df['MES_ANO'] = pd.to_datetime(df[['YEAR', 'MONTH']].assign(DAY=1))

In [7]:
df.rename(columns={'YEAR':'ANO', 'MONTH':'MÊS'}, inplace=True)

In [8]:
df.rename(columns={'MES_ANO': 'DATA'}, inplace=True)

In [9]:
df.columns

Index(['EMPRESA (SIGLA)', 'EMPRESA (NOME)', 'EMPRESA (NACIONALIDADE)', 'ANO',
       'MÊS', 'AEROPORTO DE ORIGEM (SIGLA)', 'AEROPORTO DE ORIGEM (NOME)',
       'AEROPORTO DE ORIGEM (UF)', 'AEROPORTO DE ORIGEM (REGIÃO)',
       'AEROPORTO DE ORIGEM (PAÍS)', 'AEROPORTO DE ORIGEM (CONTINENTE)',
       'AEROPORTO DE DESTINO (SIGLA)', 'AEROPORTO DE DESTINO (NOME)',
       'AEROPORTO DE DESTINO (UF)', 'AEROPORTO DE DESTINO (REGIÃO)',
       'AEROPORTO DE DESTINO (PAÍS)', 'AEROPORTO DE DESTINO (CONTINENTE)',
       'NATUREZA', 'GRUPO DE VOO', 'PASSAGEIROS PAGOS', 'PASSAGEIROS GRÁTIS',
       'CARGA PAGA (KG)', 'CARGA GRÁTIS (KG)', 'CORREIO (KG)', 'ASK', 'RPK',
       'ATK', 'RTK', 'COMBUSTÍVEL (LITROS)', 'DISTÂNCIA VOADA (KM)',
       'DECOLAGENS', 'CARGA PAGA KM', 'CARGA GRATIS KM', 'CORREIO KM',
       'ASSENTOS', 'PAYLOAD', 'HORAS VOADAS', 'BAGAGEM (KG)', 'DATA'],
      dtype='object')

In [10]:
df_europa = df[['EMPRESA (SIGLA)', 'EMPRESA (NOME)', 'EMPRESA (NACIONALIDADE)','AEROPORTO DE ORIGEM (SIGLA)', 'AEROPORTO DE ORIGEM (NOME)',
       'AEROPORTO DE ORIGEM (UF)', 'DATA', 'AEROPORTO DE ORIGEM (REGIÃO)',
       'AEROPORTO DE ORIGEM (PAÍS)', 'AEROPORTO DE ORIGEM (CONTINENTE)',
       'AEROPORTO DE DESTINO (SIGLA)', 'AEROPORTO DE DESTINO (NOME)',
       'AEROPORTO DE DESTINO (UF)', 'AEROPORTO DE DESTINO (REGIÃO)',
       'AEROPORTO DE DESTINO (PAÍS)', 'AEROPORTO DE DESTINO (CONTINENTE)',
       'NATUREZA','PASSAGEIROS PAGOS','DECOLAGENS',]]

In [11]:
df_europa2 = df_europa.copy()

In [12]:
df_europa2.drop(df_europa2[df_europa2['NATUREZA'] != 'INTERNACIONAL'].index, axis=0, inplace=True)
df_europa2.drop(df_europa2[['AEROPORTO DE ORIGEM (UF)','AEROPORTO DE ORIGEM (REGIÃO)']], axis=1, inplace=True)
df_europa2.drop(df_europa2[df_europa2['AEROPORTO DE ORIGEM (CONTINENTE)'] != 'EUROPA'].index, axis=0, inplace=True)
df_europa2.drop(df_europa2[df_europa2['AEROPORTO DE DESTINO (PAÍS)'] != 'BRASIL'].index, axis=0, inplace=True)
df_europa2.drop(df_europa2[df_europa2['PASSAGEIROS PAGOS'] == 0].index, axis=0, inplace=True)

In [13]:
df_europa2.isnull().sum()

EMPRESA (SIGLA)                         0
EMPRESA (NOME)                          0
EMPRESA (NACIONALIDADE)                 0
AEROPORTO DE ORIGEM (SIGLA)             0
AEROPORTO DE ORIGEM (NOME)              0
DATA                                    0
AEROPORTO DE ORIGEM (PAÍS)              0
AEROPORTO DE ORIGEM (CONTINENTE)        0
AEROPORTO DE DESTINO (SIGLA)            0
AEROPORTO DE DESTINO (NOME)             0
AEROPORTO DE DESTINO (UF)               0
AEROPORTO DE DESTINO (REGIÃO)           0
AEROPORTO DE DESTINO (PAÍS)             0
AEROPORTO DE DESTINO (CONTINENTE)       0
NATUREZA                                0
PASSAGEIROS PAGOS                     213
DECOLAGENS                           1510
dtype: int64

## Breakpoint
As can be seen, the column “DECOLAGENS” (means Take-off in Portuguese) has 1510 empty values. In this case, for the purpose of our analysis, we can set all those fields with number one.  Since each line in the dataset represents the number of flights in each segment flown to Brazil, we can assume that, at least, one flight occurred for each segment, otherwise it would not appear in the official records.
For the column “PASSAGEIROS PAGOS” (“paying passengers” in English), we can drop all the rows, because if there were no passengers, it means that they were cargo flights, which is not relevant for the analysis.


In [14]:
df_europa2['DECOLAGENS'].fillna(value=1, inplace=True)

In [15]:
df_europa2.dropna(subset =['PASSAGEIROS PAGOS'], axis=0, inplace=True)

In [16]:
df_europa2.isnull().sum()

EMPRESA (SIGLA)                      0
EMPRESA (NOME)                       0
EMPRESA (NACIONALIDADE)              0
AEROPORTO DE ORIGEM (SIGLA)          0
AEROPORTO DE ORIGEM (NOME)           0
DATA                                 0
AEROPORTO DE ORIGEM (PAÍS)           0
AEROPORTO DE ORIGEM (CONTINENTE)     0
AEROPORTO DE DESTINO (SIGLA)         0
AEROPORTO DE DESTINO (NOME)          0
AEROPORTO DE DESTINO (UF)            0
AEROPORTO DE DESTINO (REGIÃO)        0
AEROPORTO DE DESTINO (PAÍS)          0
AEROPORTO DE DESTINO (CONTINENTE)    0
NATUREZA                             0
PASSAGEIROS PAGOS                    0
DECOLAGENS                           0
dtype: int64

In [17]:
df_europa2.sample(5)

EMPRESA (SIGLA)                                     EMPRESA (NOME)  \
509676             TAP           TAP - TRANSPORTES AÉREOS PORTUGUESES S/A   
527568             CFG                             CONDOR FLUGDIENST GMBH   
442798             IBE  IBÉRIA LINEAS AEREAS DE ESPAÑA SOCIEDAD ANONIM...   
745474             TAP           TAP - TRANSPORTES AÉREOS PORTUGUESES S/A   
769873             KLM                 KLM CIA. REAL HOLANDESA DE AVIAÇÃO   

       EMPRESA (NACIONALIDADE) AEROPORTO DE ORIGEM (SIGLA)  \
509676             ESTRANGEIRA                        LPPR   
527568             ESTRANGEIRA                        EDDF   
442798             ESTRANGEIRA                        LEMD   
745474             ESTRANGEIRA                        LPPR   
769873             ESTRANGEIRA                        EHAM   

            AEROPORTO DE ORIGEM (NOME)       DATA AEROPORTO DE ORIGEM (PAÍS)  \
509676                           PORTO 2011-06-01                   PORTUGAL   
527568               FRANKFURT AM MAIN 2012-08-01                   ALEMANHA   
442798                          MADRID 2010-09-01                    ESPANHA   
745474                           PORTO 2016-04-01                   PORTUGAL   
769873  HAARLEMMERMEER, NEAR AMSTERDAM 2017-11-01                    HOLANDA   

       AEROPORTO DE ORIGEM (CONTINENTE) AEROPORTO DE DESTINO (SIGLA)  \
509676                           EUROPA                         SBGR   
527568                           EUROPA                         SBSV   
442798                           EUROPA                         SBGL   
745474                           EUROPA                         SBGR   
769873                           EUROPA                         SBGL   

       AEROPORTO DE DESTINO (NOME) AEROPORTO DE DESTINO (UF)  \
509676                   GUARULHOS                        SP   
527568                    SALVADOR                        BA   
442798              RIO DE JANEIRO                        RJ   
745474                   GUARULHOS                        SP   
769873              RIO DE JANEIRO                        RJ   

       AEROPORTO DE DESTINO (REGIÃO) AEROPORTO DE DESTINO (PAÍS)  \
509676                       SUDESTE                      BRASIL   
527568                      NORDESTE                      BRASIL   
442798                       SUDESTE                      BRASIL   
745474                       SUDESTE                      BRASIL   
769873                       SUDESTE                      BRASIL   

       AEROPORTO DE DESTINO (CONTINENTE)       NATUREZA  PASSAGEIROS PAGOS  \
509676                    AMÉRICA DO SUL  INTERNACIONAL             1711.0   
527568                    AMÉRICA DO SUL  INTERNACIONAL             2113.0   
442798                    AMÉRICA DO SUL  INTERNACIONAL             7541.0   
745474                    AMÉRICA DO SUL  INTERNACIONAL              466.0   
769873                    AMÉRICA DO SUL  INTERNACIONAL             7163.0   

        DECOLAGENS  
509676         8.0  
527568         9.0  
442798        30.0  
745474         4.0  
769873        26.0

In [18]:
df_europa2.to_csv('Voos_Europa_sem_ICAO.csv', index= False)

In [19]:
airport_codes_original = pd.read_csv('airport-codes_csv.csv')

In [20]:
airport_codes_original.sample(3)

ident           type                        name  elevation_ft  \
44993  SIXX  small_airport  Usinas Braco Norte Airport        1159.0   
2737   25NY       heliport           Carter's Heliport           5.0   
16100  CNQ4  small_airport         Tillsonburg Airport         894.0   

      continent iso_country iso_region        municipality gps_code iata_code  \
44993        SA          BR      BR-MT  Guarantã Do Norte     SIXX       NaN   
2737        NaN          US      US-NY          Remsenburg     25NY       NaN   
16100       NaN          CA      CA-ON         Tillsonburg     CYTB       NaN   

      local_code                              coordinates  
44993        NaN  -54.954444885253906, -9.658332824707031  
2737        25NY    -72.69950103759766, 40.80009841918945  
16100       CYTB                      -80.746944, 42.9275

In [21]:
airport_codes_original.columns

Index(['ident', 'type', 'name', 'elevation_ft', 'continent', 'iso_country',
       'iso_region', 'municipality', 'gps_code', 'iata_code', 'local_code',
       'coordinates'],
      dtype='object')

In [26]:
airport_codes_novo = airport_codes_original[['ident', 'coordinates']].copy()
airport_codes_novo.sample()

ident                       coordinates
18910  EDSG  9.436944007870002, 48.5361099243

In [27]:
airport_codes_novo[['Longitude','Latitude']] = airport_codes_novo['coordinates'].str.split(',',expand=True)

In [28]:
airport_codes_novo.sample(10)

ident                               coordinates            Longitude  \
12055  AZ-0018               48.9667510986, 39.831790367        48.9667510986   
30043     KPTS     -94.7311019897461, 37.449501037597656    -94.7311019897461   
22885  GB-0167                      -2.920267, 55.005191            -2.920267   
49018     SWOZ  -54.899166107177734, -15.310556411743164  -54.899166107177734   
21377     FLAT     32.06439971923828, -14.12030029296875    32.06439971923828   
42540     SACO                       -64.208, -31.323601              -64.208   
39457     OOYB    56.034400939941406, 22.201499938964844   56.034400939941406   
38684     NZRK    176.36399841308594, -38.88639831542969   176.36399841308594   
17739  DE-0069                          11.6667, 47.8519              11.6667   
52902     VA09      -82.58489990234375, 36.9718017578125   -82.58489990234375   

                   Latitude  
12055          39.831790367  
30043    37.449501037597656  
22885             55.005191  
49018   -15.310556411743164  
21377    -14.12030029296875  
42540            -31.323601  
39457    22.201499938964844  
38684    -38.88639831542969  
17739               47.8519  
52902      36.9718017578125

In [29]:
airport_codes_novo.drop(airport_codes_novo[['coordinates']], axis=1, inplace=True)

In [30]:
airport_codes_novo.sample(3)

ident   Longitude     Latitude
13469  BR-0991  -46.701765   -23.624984
12863  BR-0385  -35.497589    -8.187697
42314  RU-0512   40.157771    56.296683

In [31]:
airport_codes_novo.to_csv('ICAO_long_lat.csv', index=False)

In [32]:
df_europa2.sample(2)

EMPRESA (SIGLA)                      EMPRESA (NOME)  \
756567             AZU  AZUL LINHAS AÉREAS BRASILEIRAS S/A   
259837             VRG  VARIG - VIAÇÃO AÉREA RIO-GRANDENSE   

       EMPRESA (NACIONALIDADE) AEROPORTO DE ORIGEM (SIGLA)  \
756567              BRASILEIRA                        LPPT   
259837              BRASILEIRA                        EHAM   

            AEROPORTO DE ORIGEM (NOME)       DATA AEROPORTO DE ORIGEM (PAÍS)  \
756567                          LISBOA 2017-10-01                   PORTUGAL   
259837  HAARLEMMERMEER, NEAR AMSTERDAM 2005-09-01                    HOLANDA   

       AEROPORTO DE ORIGEM (CONTINENTE) AEROPORTO DE DESTINO (SIGLA)  \
756567                           EUROPA                         SBKP   
259837                           EUROPA                         SBGR   

       AEROPORTO DE DESTINO (NOME) AEROPORTO DE DESTINO (UF)  \
756567                    CAMPINAS                        SP   
259837                   GUARULHOS                        SP   

       AEROPORTO DE DESTINO (REGIÃO) AEROPORTO DE DESTINO (PAÍS)  \
756567                       SUDESTE                      BRASIL   
259837                       SUDESTE                      BRASIL   

       AEROPORTO DE DESTINO (CONTINENTE)       NATUREZA  PASSAGEIROS PAGOS  \
756567                    AMÉRICA DO SUL  INTERNACIONAL              244.0   
259837                    AMÉRICA DO SUL  INTERNACIONAL             1125.0   

        DECOLAGENS  
756567         1.0  
259837         1.0

In [33]:
df_Europa_sem_lat_lon = df_europa2.merge(airport_codes_novo, left_on="AEROPORTO DE ORIGEM (SIGLA)", right_on='ident', how='left')

In [34]:
df_Europa_sem_lat_lon.sample()

EMPRESA (SIGLA)                      EMPRESA (NOME)  \
2505             VRG  VARIG - VIAÇÃO AÉREA RIO-GRANDENSE   

     EMPRESA (NACIONALIDADE) AEROPORTO DE ORIGEM (SIGLA)  \
2505              BRASILEIRA                        LEMD   

     AEROPORTO DE ORIGEM (NOME)       DATA AEROPORTO DE ORIGEM (PAÍS)  \
2505                     MADRID 2003-09-01                    ESPANHA   

     AEROPORTO DE ORIGEM (CONTINENTE) AEROPORTO DE DESTINO (SIGLA)  \
2505                           EUROPA                         SBGR   

     AEROPORTO DE DESTINO (NOME) AEROPORTO DE DESTINO (UF)  \
2505                   GUARULHOS                        SP   

     AEROPORTO DE DESTINO (REGIÃO) AEROPORTO DE DESTINO (PAÍS)  \
2505                       SUDESTE                      BRASIL   

     AEROPORTO DE DESTINO (CONTINENTE)       NATUREZA  PASSAGEIROS PAGOS  \
2505                    AMÉRICA DO SUL  INTERNACIONAL              127.0   

      DECOLAGENS ident Longitude    Latitude  
2505         1.0  LEMD  -3.56264   40.471926

In [35]:
df_Europa_sem_lat_lon.rename(columns={'Longitude':'LONGITUDE DE ORIGEM', 'Latitude':'LATITUDE DE ORIGEM'}, inplace=True)
df_Europa_sem_lat_lon.sample()

EMPRESA (SIGLA)                                     EMPRESA (NOME)  \
12058             IBE  IBÉRIA LINEAS AEREAS DE ESPAÑA SOCIEDAD ANONIM...   

      EMPRESA (NACIONALIDADE) AEROPORTO DE ORIGEM (SIGLA)  \
12058             ESTRANGEIRA                        LEMD   

      AEROPORTO DE ORIGEM (NOME)       DATA AEROPORTO DE ORIGEM (PAÍS)  \
12058                     MADRID 2019-04-01                    ESPANHA   

      AEROPORTO DE ORIGEM (CONTINENTE) AEROPORTO DE DESTINO (SIGLA)  \
12058                           EUROPA                         SBGL   

      AEROPORTO DE DESTINO (NOME) AEROPORTO DE DESTINO (UF)  \
12058              RIO DE JANEIRO                        RJ   

      AEROPORTO DE DESTINO (REGIÃO) AEROPORTO DE DESTINO (PAÍS)  \
12058                       SUDESTE                      BRASIL   

      AEROPORTO DE DESTINO (CONTINENTE)       NATUREZA  PASSAGEIROS PAGOS  \
12058                    AMÉRICA DO SUL  INTERNACIONAL             4645.0   

       DECOLAGENS ident LONGITUDE DE ORIGEM LATITUDE DE ORIGEM  
12058        21.0  LEMD            -3.56264          40.471926

In [36]:
Dataframe_Europa_completo = df_Europa_sem_lat_lon.merge(airport_codes_novo, left_on="AEROPORTO DE DESTINO (SIGLA)", right_on='ident', how='left')
Dataframe_Europa_completo.sample()

EMPRESA (SIGLA)          EMPRESA (NOME) EMPRESA (NACIONALIDADE)  \
12606             TAM  TAM LINHAS AÉREAS S.A.              BRASILEIRA   

      AEROPORTO DE ORIGEM (SIGLA) AEROPORTO DE ORIGEM (NOME)       DATA  \
12606                        EGYP           FALKLAND ISLANDS 2020-01-01   

      AEROPORTO DE ORIGEM (PAÍS) AEROPORTO DE ORIGEM (CONTINENTE)  \
12606                REINO UNIDO                           EUROPA   

      AEROPORTO DE DESTINO (SIGLA) AEROPORTO DE DESTINO (NOME)  ...  \
12606                         SBGR                   GUARULHOS  ...   

      AEROPORTO DE DESTINO (CONTINENTE)       NATUREZA PASSAGEIROS PAGOS  \
12606                    AMÉRICA DO SUL  INTERNACIONAL              92.0   

      DECOLAGENS ident_x  LONGITUDE DE ORIGEM   LATITUDE DE ORIGEM ident_y  \
12606        4.0    EGYP  -58.447200775146484   -51.82279968261719    SBGR   

                Longitude              Latitude  
12606  -46.47305679321289   -23.435556411743164  

[1 rows x 23 columns]

In [37]:
Dataframe_Europa_completo.rename(columns={'Longitude':'LONGITUDE DE DESTINO', 'Latitude':'LATITUDE DE DESTINO'}, inplace=True)
Dataframe_Europa_completo.sample()

EMPRESA (SIGLA)       EMPRESA (NOME) EMPRESA (NACIONALIDADE)  \
6862             BAW  BRITISH AIRWAYS PLC             ESTRANGEIRA   

     AEROPORTO DE ORIGEM (SIGLA) AEROPORTO DE ORIGEM (NOME)       DATA  \
6862                        EGLL            LONDON, ENGLAND 2011-12-01   

     AEROPORTO DE ORIGEM (PAÍS) AEROPORTO DE ORIGEM (CONTINENTE)  \
6862                REINO UNIDO                           EUROPA   

     AEROPORTO DE DESTINO (SIGLA) AEROPORTO DE DESTINO (NOME)  ...  \
6862                         SBGL              RIO DE JANEIRO  ...   

     AEROPORTO DE DESTINO (CONTINENTE)       NATUREZA PASSAGEIROS PAGOS  \
6862                    AMÉRICA DO SUL  INTERNACIONAL            5724.0   

     DECOLAGENS ident_x  LONGITUDE DE ORIGEM  LATITUDE DE ORIGEM ident_y  \
6862       27.0    EGLL            -0.461941             51.4706    SBGL   

     LONGITUDE DE DESTINO LATITUDE DE DESTINO  
6862       -43.2505569458      -22.8099994659  

[1 rows x 23 columns]

In [38]:
Dataframe_Europa_completo.drop(Dataframe_Europa_completo[['ident_x', 'ident_y']], axis=1, inplace=True)
Dataframe_Europa_completo.sample(3)

EMPRESA (SIGLA)                             EMPRESA (NOME)  \
5124              AFR                         SOCIÉTÉ AIR FRANCE   
9050              TAM                     TAM LINHAS AÉREAS S.A.   
11840             AEA  AIR EUROPA LINEAS AEREAS SOCIEDAD ANONIMA   

      EMPRESA (NACIONALIDADE) AEROPORTO DE ORIGEM (SIGLA)  \
5124              ESTRANGEIRA                        LFPG   
9050               BRASILEIRA                        LFPG   
11840             ESTRANGEIRA                        LEMD   

      AEROPORTO DE ORIGEM (NOME)       DATA AEROPORTO DE ORIGEM (PAÍS)  \
5124                       PARIS 2008-05-01                     FRANÇA   
9050                       PARIS 2014-07-01                     FRANÇA   
11840                     MADRID 2019-09-01                    ESPANHA   

      AEROPORTO DE ORIGEM (CONTINENTE) AEROPORTO DE DESTINO (SIGLA)  \
5124                            EUROPA                         SBGR   
9050                            EUROPA                         SBGR   
11840                           EUROPA                         SBSV   

      AEROPORTO DE DESTINO (NOME)  ... AEROPORTO DE DESTINO (REGIÃO)  \
5124                    GUARULHOS  ...                       SUDESTE   
9050                    GUARULHOS  ...                       SUDESTE   
11840                    SALVADOR  ...                      NORDESTE   

      AEROPORTO DE DESTINO (PAÍS) AEROPORTO DE DESTINO (CONTINENTE)  \
5124                       BRASIL                    AMÉRICA DO SUL   
9050                       BRASIL                    AMÉRICA DO SUL   
11840                      BRASIL                    AMÉRICA DO SUL   

            NATUREZA PASSAGEIROS PAGOS  DECOLAGENS  LONGITUDE DE ORIGEM  \
5124   INTERNACIONAL           11699.0        67.0                 2.55   
9050   INTERNACIONAL           10643.0        31.0                 2.55   
11840  INTERNACIONAL            1722.0         8.0             -3.56264   

      LATITUDE DE ORIGEM LONGITUDE DE DESTINO   LATITUDE DE DESTINO  
5124           49.012798   -46.47305679321289   -23.435556411743164  
9050           49.012798   -46.47305679321289   -23.435556411743164  
11840          40.471926       -38.3224983215        -12.9086112976  

[3 rows x 21 columns]

In [39]:
Dataframe_Europa_completo.to_csv('Dataframe_Europa_completo.csv', index=False)